src/server/ftpserver.rs

```diff
@@ -21,9 +21,27 @@ use std::time::Duration;
 use tokio::sync::Mutex;
 use tokio_util::codec::*;
 
+use std::net::{IpAddr, SocketAddr};
+
 const DEFAULT_GREETING: &str = "Welcome to the libunftp FTP server";
 const DEFAULT_IDLE_SESSION_TIMEOUT_SECS: u64 = 600;
 
+#[derive(Clone, Copy)]
```

Comment:
After I read [the API Guidelines](https://rust-lang.github.io/api-guidelines/interoperability.html#c-common-traits) I'd recommend we get into the habit to implement Display and Debug etc. too.